### PR TITLE
Update election based lifecycling docs

### PR DESCRIPTION
**Pull request by markus-da**
Fixes https://github.com/digital-asset/daml-finance/issues/759

Also fixes parts of https://github.com/digital-asset/daml-finance/issues/749

### CODE REVIEW

File: docs/2.7.0/docs/daml-finance/tutorials/instrument-modeling/generic-extension.rst [hunk 1]
```diff
@@ -128,22 +133,37 @@ It is now possible to create the *Election*:
   :start-after: -- CREATE_ELECTION_BEGIN
   :end-before: -- CREATE_ELECTION_END
 
+Note: these parts (election offer and election candidate) are not considered a core part of the Daml
```
Comment (matteolimberto-da) [inline docs/2.7.0/docs/daml-finance/tutorials/instrument-modeling/generic-extension.rst:136]:
```suggestion
Note: these templates (election offer and election candidate) are not considered a core part of the Daml
```

File: docs/2.7.0/docs/daml-finance/tutorials/instrument-modeling/generic-extension.rst [hunk 2]
```diff
@@ -128,22 +133,37 @@ It is now possible to create the *Election*:
...
 
 .. literalinclude:: ../../src/test/daml/Daml/Finance/Instrument/Generic/Test/EuropeanOption.daml
   :language: daml
   :start-after: -- APPLY_ELECTION_BEGIN
   :end-before: -- APPLY_ELECTION_END
 
-This create lifecycle effects, which can be claimed and settled in the usual way. This is described
-in :doc:`Getting Started: Lifecycling <../getting-started/lifecycling>`.
+This creates lifecycle effects, which can be claimed and settled in the usual way (as described in
+:doc:`Getting Started: Lifecycling <../getting-started/lifecycling>`). The value of *electorIsOwner*
```
Comment (matteolimberto-da) [inline docs/2.7.0/docs/daml-finance/tutorials/instrument-modeling/generic-extension.rst:165]: I don't think that's the case? 

The flag just controls which party makes the election, but the effects are generated through lifecycling (where the `electorIsOwner` flag does not play any role

**Review by matteolimberto-da — CHANGES_REQUESTED** [review]
See comment

File: docs/2.7.0/docs/daml-finance/tutorials/instrument-modeling/generic-extension.rst (hunk 2 above)
Comment (markus-da) [reply docs/2.7.0/docs/daml-finance/tutorials/instrument-modeling/generic-extension.rst:165]: If it is misleading/incorrect, I could take it out.
Or you could propose an alternative formulation, as you prefer.

Comment (matteolimberto-da) [reply docs/2.7.0/docs/daml-finance/tutorials/instrument-modeling/generic-extension.rst:165]: I could suggest the following

This creates lifecycle effects, which can be claimed and settled in the usual way (as described in
:doc:`Getting Started: Lifecycling <../getting-started/lifecycling>`).  However,  the holding contract used to claim the effect must be compatible with the election that has been made: if `alice` made an election and `electorIsOwner = True`, then only a holding where `owner = alice` will be accepted.

If you think it makes the description more confusing, then I would take out the last sentence and stop at "can be claimed in the usual way"

Comment (markus-da) [reply docs/2.7.0/docs/daml-finance/tutorials/instrument-modeling/generic-extension.rst]: I think this makes sense. I have changed it accordingly.

**Review by matteolimberto-da — APPROVED** [review]
LGTM

**Pull request merged**